File: EcuacionLineal_Java/Archivos/pruebaArchivos.java
package Archivos;
import java.io.*;
import java.util.ArrayList;

public class pruebaArchivos {
    private ArrayList<String> contenidoArchivo = new ArrayList<>();
    private ArrayList<Ecuacion> listaEntrada = new ArrayList<>();
    private final String rutaWindows =  "C:\\Users\\Erick\\Documents\\Programacion\\Practica 1.1\\src\\Archivos\\datos.txt";

    public ArrayList<String> getContenidoArchivo() {
        return contenidoArchivo;
    }

    public ArrayList<Ecuacion> getListaEntrada() {
        return listaEntrada;
    }

    public void leerArchivo() {
        LeerArchivo leer = new LeerArchivo();
        this.contenidoArchivo =
                leer.leerArchivo(this.rutaWindows);
        for (int i = 0; i < this.contenidoArchivo.size(); i++) {
            String linea = this.contenidoArchivo.get(i);
            String elementosLinea[] = linea.split(";");
            try {
                int coeficienteA = Integer.parseInt(elementosLinea[0]);
                int coeficienteB = Integer.parseInt(elementosLinea[1]);
                int terminoIndependienteC = Integer.parseInt(elementosLinea[2]);
                int coeficienteD = Integer.parseInt(elementosLinea[3]);
                int coeficienteE = Integer.parseInt(elementosLinea[4]);
                int terminoIndependienteF = Integer.parseInt(elementosLinea[5]);
                listaEntrada.add(new Ecuacion(coeficienteA, coeficienteB, terminoIndependienteC, coeficienteD, coeficienteE, terminoIndependienteF));
            } catch (NumberFormatException ex) {
                ex.printStackTrace();
                listaEntrada.add(new Ecuacion(0, 0, 0, 0, 0, 0));
            }
        }
    }
    public void escribirArchivo(ArrayList<Ecuacion> lista){
        String archivo = "resultados.csv";
        File f = new File(archivo);
        try{
            FileWriter w = new FileWriter(f);
            BufferedWriter bw = new BufferedWriter(w);
            PrintWriter wr = new PrintWriter(bw);
            wr.write("");
            wr.append("coeficienteA, coeficienteB, terminoIndependienteC, coeficienteD, coeficienteE, terminoIndependienteF, variableX, variableY\n");
            for (int i=0; i < lista.size(); i++){
                wr.append(lista.get(i).getCoeficienteA() + "," + lista.get(i).getCoeficienteB() +"," + lista.get(i).getTerminoIndependienteC() +
                        "," + lista.get(i).getCoeficienteD() + "," + lista.get(i).getCoeficienteE() + "," + lista.get(i).getTerminoIndependienteF() +
                        "," + lista.get(i).getVariableX() + "," + lista.get(i).getVariableY() +  "\n");
            }
            wr.close();
            bw.close();
        }catch(IOException e){

        }
    }

    public static void main (String args[]){
        pruebaArchivos prueba = new pruebaArchivos();
        prueba.leerArchivo();
        for (int i=0; i < prueba.getListaEntrada().size(); i++){
            System.out.println(prueba.getListaEntrada().get(i).getCoeficienteA() +
                    "\t" + prueba.getListaEntrada().get(i).getCoeficienteB() +
                    "\t" + prueba.getListaEntrada().get(i).getTerminoIndependienteC() +
                    "\t" + prueba.getListaEntrada().get(i).getCoeficienteD() +
                    "\t" + prueba.getListaEntrada().get(i).getCoeficienteE() +
                    "\t" + prueba.getListaEntrada().get(i).getTerminoIndependienteF() +
                    "\t" + prueba.getListaEntrada().get(i).getVariableX() +
                    "\t" + prueba.getListaEntrada().get(i).getVariableY());
        }
        prueba.escribirArchivo(prueba.getListaEntrada());
    }
}

File: EcuacionLineal_Python/Arreglos/pruebaArchivos.py
from abc import ABCMeta


class Ecuacion(ABCMeta):

    # Metodo de clase
    @staticmethod
    def calcularIncognitas(coeficienteA, coeficienteB, terminoIndependienteC, coeficienteD, coeficienteE,
                           terminoIndependienteF):
        determinante = coeficienteA * coeficienteE - coeficienteB * coeficienteD
        variableX = coeficienteE * terminoIndependienteC - coeficienteB * terminoIndependienteF / determinante
        variableX = round(variableX, 2)
        variableY = coeficienteA * terminoIndependienteF - coeficienteD * terminoIndependienteC / determinante
        variableY = round(variableY, 2)
        return [variableX, variableY]


class pruebaArchivos:

    def leerArchivo(self, archivo):
        file = open(archivo, 'r')
        lineas = []
        lineas_archivo = []
        for linea in file.readlines():
            lineas.append(linea.replace('\n', '').split(";"))
        file.close()
        for f in range(0, len(lineas)):
            try:
                lineas_archivo.append(
                    [int(lineas[f][0]), int(lineas[f][1]), int(lineas[f][2]), int(lineas[f][3]), int(lineas[f][4]),
                     int(lineas[f][5])])
            except ValueError:
                lineas_archivo.append([0, 0, 0, 0, 0, 0])
        return lineas_archivo

    def Variables(self, lista):
        resultados = []
        for f in range(0, len(lista)):
            resultados.append(Ecuacion.calcularIncognitas(
                lista[f][0], lista[f][1], lista[f][2], lista[f][3], lista[f][4], lista[f][5]))
        return resultados

    def guardarResultados(self, entrada, resultados):
        file = open("resultados.txt", 'w')
        file.write('CoeficienteA;CoeficienteB;TerminoIndependienteC;CoeficienteD;CoeficienteE;TerminoIndependienteF;'
                   ' VariableX; VariableY\n')
        for f in range(0, len(entrada)):
            linea = str(entrada[f][0]) + ';' + str(entrada[f][1]) + ';' + str(entrada[f][2]) + ';' + str(
                entrada[f][3]) + ';' + str(entrada[f][4]) + ';' + str(entrada[f][5]) + ';' + str(resultados[f][0]) + ';'+ \
                str(resultados[f][1]) + '\n'
            file.write(linea)
        file.close()


if __name__ == "__main__":
    prueba = pruebaArchivos()
    archivo = []
    archivo = prueba.leerArchivo("datos.txt")
    resultado = prueba.Variables(archivo)
    prueba.guardarResultados(archivo, resultado)
